components/event-chart/src/event-chart.ts

```diff
@@ -474,6 +470,14 @@ export class DtEventChart<T> implements AfterContentInit, OnInit, OnDestroy {
     }
   }
 
+  /**
+   * Track by function for the renderEvents – speeds up performance
+   * and prevent deletion of mouseout
+   */
+  _renderEventTrackByFn(index: number, _item: RenderEvent<T>): number {
+    return index;
+  }
```

Comment:
You are tracking them by their array index, which does not work. If the values within the renderevents change, you will need to rerender them. Please check them based on either their originalIndex, or a combination of index and value.

Reply:
The index is only for identifying and it is working perfectly this way – take a look at this stackblitz where I demonstrate the behaviour:

https://stackblitz.com/edit/angular-prnpnp?file=src%2Fapp%2Fapp.component.ts